benchmarks/solubility/plugin/solubility.cc

```diff
@@ -0,0 +1,409 @@
+          reference_permeability            = prm.get_double ("Reference permeability");
+          alpha_phi                         = prm.get_double ("Exponential melt weakening factor");
+          melt_compressibility              = prm.get_double ("Melt compressibility");
+          melting_time_scale                = prm.get_double ("Melting time scale for operator splitting");
+
+          // Create the base model and initialize its SimulatorAccess base
+          // class; it will get a chance to read its parameters below after we
+          // leave the current section.
+          base_model = create_material_model<dim>(prm.get("Base model"));
+          if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(base_model.get()))
+            sim->initialize_simulator (this->get_simulator());
+
+          if (this->convert_output_to_years() == true)
+            melting_time_scale *= year_in_seconds;
+
+          if (this->get_parameters().use_operator_splitting)
+            {
+              AssertThrow(melting_time_scale >= this->get_parameters().reaction_time_step,
+                          ExcMessage("The reaction time step " + Utilities::to_string(this->get_parameters().reaction_time_step)
+                                     + " in the operator splitting scheme is too large to compute melting rates! "
+                                     "You have to choose it in such a way that it is smaller than the 'Melting time scale for "
+                                     "operator splitting' chosen in the material model, which is currently "
+                                     + Utilities::to_string(melting_time_scale) + "."));
+              AssertThrow(melting_time_scale > 0,
+                          ExcMessage("The Melting time scale for operator splitting must be larger than 0!"));
+            }
+
+          AssertThrow(this->introspection().compositional_name_exists("porosity"),
+                      ExcMessage("Material model Volatiles only "
+                                 "works if there is a compositional field called porosity."));
+
+          AssertThrow(this->introspection().compositional_name_exists("water_content"),
+                      ExcMessage("Material model Volatiles only "
+                                 "works if there is a compositional field called water_content."));
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      // After parsing the parameters for this model, parse parameters related to the base model.
+      base_model->parse_parameters(prm);
+      this->model_dependence = base_model->get_model_dependence();
+    }
+
+
+
+    template <int dim>
+    bool
+    Volatiles<dim>::
+    is_compressible () const
+    {
+      return base_model->is_compressible();
+    }
+
+
+
+    template <int dim>
+    void
+    Volatiles<dim>::
+    melt_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
```

Comment:
```suggestion
    fluid_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
```

Reply:
Same here, this is not something I can change, this is defined by the interface.